oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/tools/IndexComparator.java

```diff
@@ -0,0 +1,247 @@
+                // ".si" files contain a timestamp
+                ignoreChecksums = true;
+            } else if (path.endsWith(".cfs")
+                    || path.endsWith(".cfe")) {
+                // ".cfs" and ".cfe files depends on
+                // the aggregation order
+                ignoreChecksums = true;
+            }
+        }
+        Set<String> both = new HashSet<>(o1.getProperties().keySet());
+        both.addAll(o2.getProperties().keySet());
+        for (String k : both) {
+            String v1 = o1.getProperties().get(k);
+            String v2 = o2.getProperties().get(k);
+            if (v1 == null || v2 == null || !v1.equals(v2)) {
+                if (ignoreChecksums) {
+                    if ("sha256".equals(k) || "md5".equals(k)) {
+                        continue;
+                    }
+                    if ("bytes".equals(k) && v1 != null && v2 != null) {
+                        long x1 = Long.parseLong(v1);
+                        long x2 = Long.parseLong(v2);
+                        if (Math.abs(x1 - x2) < 1000) {
+                            // less than 1000 bytes difference
+                            // if aggregation order is different
+                            continue;
+                        } else if (Math.abs((double) x1 / x2 - 1.0) < 0.001) {
+                            // less than 0.1% difference
+                            // if aggregation order is different
+                            continue;
+                        }
+                    }
+                }
+                result.key(path + "/" + k).object().
+                    key("v1").encodedValue(v1).
+                    key("v2").encodedValue(v2).
+                    endObject();
+            }
+        }
+        both = new HashSet<>(o1.getChildren().keySet());
+        both.addAll(o2.getChildren().keySet());
+        for (String k : both) {
+            JsonObject v1 = o1.getChildren().get(k);
+            JsonObject v2 = o2.getChildren().get(k);
+            if (v1 == null || v2 == null) {
+                result.key(path + "/" + k).object().
+                    key("v1").encodedValue("" + v1).
+                    key("v2").encodedValue("" + v2).
+                    endObject();
+            } else {
+                diff(v1, v2, path + "/" + k, result);
+            }
+        }
+    }
+
+    /**
+     * Create or load a string that contains the JSON representation of the
+     * statistics of an index.
+     *
+     * @param file the Lucene directory, or JSON file
+     * @return the statistics in JSON form
+     */
+    private static String getStats(File file) throws Exception {
+        if (file.isDirectory()) {
+            // comparing the directory
+            return getStatsFromLucene(file);
+        } else {
+            // reading a pre-existing (json) file that contains
+            // the statistics
+            return new String(Files.readAllBytes(file.toPath()), StandardCharsets.UTF_8);
```

Comment:
this can be replaced with 

```suggestion
            return Files.readString(file.toPath());
```